sources2csr/sources_reader.py

```diff
@@ -43,330 +35,149 @@ def transform_entity(values: Dict[Any, Any], schema: Dict) -> Dict:
             for k, v in values.items()}
 
 
-def transform_entities(entities_df: Any, schema: Dict, constructor: Any):
-    entities = [transform_entity(entity, schema) for entity in entities_df.to_dict(orient='records')]
+def transform_entities(entities: Any, schema: Dict, constructor: Any):
+    entities = [transform_entity(entity, schema) for entity in entities]
     return [constructor(entity) for entity in entities]
 
 
-class SourcesReader:
+def read_configuration(config_dir) -> SourcesConfig:
+    """ Parse configuration files and return set of dictionaries
 
-    def __init__(self, input_dir, output_dir, config_dir):
-        self.input_dir = input_dir
-        self.output_dir = output_dir
-        self.file_prop_dict = None
-        self.columns_to_csr_map = None
-        self.column_prio_dict = None
-        self.csr_data_model = None
-        self.study_data_model = None
-        self.expected_files = None
-        self.read_configuration(config_dir)
-
-    def read_configuration(self, config_dir):
-        """ Parse configuration files and return set of dictionaries
-
-        :param config_dir: Path to directory where the configs are stored
-        """
+    :param config_dir: Path to directory where the configs are stored
+    """
+    sources_config_path = path.join(config_dir, 'sources_config.json')
+    if not path.exists(sources_config_path) or not path.isfile(sources_config_path):
+        raise DataException(f'Cannot find {sources_config_path}')
+    with open(sources_config_path, 'r') as sources_config_file:
+        return SourcesConfig(**json.load(sources_config_file))
 
-        data_model_ = read_dict_from_file(filename='data_model.json', path=config_dir)
-        self.file_prop_dict = read_dict_from_file(filename='file_headers.json', path=config_dir)
-        self.columns_to_csr_map = read_dict_from_file(filename='columns_to_csr.json', path=config_dir)
-        self.column_prio_dict = read_dict_from_file(filename='column_priority.json', path=config_dir)
 
-        if 'central_subject_registry' in data_model_:
-            self.csr_data_model = data_model_['central_subject_registry']
-        else:
-            logger.error('"central_subject_registry" object missing from the data_model.json file, exiting')
-            raise DataException()
-        if 'study_registry' in data_model_:
-            self.study_data_model = data_model_['study_registry']
+def read_tsv_file(source_file) -> Sequence[Dict[str, Any]]:
```

Comment:
Could it be a part of `TsvReader`? Then it could be reused in csr.utils.

Reply:
Good point.